If-Statements/grade_to_points/grade_to_points.rs

```diff
@@ -0,0 +1,51 @@
+use std::io;
+
+fn main() {
+    let mut grade = String::new();
+
+    println!("Please enter an American letter grade. (e.g, B-)");
+    io::stdin().read_line(&mut grade)
+        .expect("Failed to read line.");
+
+    let grade = grade.trim().to_uppercase();
+    let points;
+
+    if grade == "A+" {
+        points = 4.0;
+    } else if grade == "A" {
+        points = 4.0;
+    } else if grade == "A-" {
+        points = 3.7;
+    } else if grade == "B+" {
+        points = 3.3;
+    } else if grade == "B" {
+        points = 3.0;
+    } else if grade == "B-" {
+        points = 2.7;
+    } else if grade == "C+" {
+        points = 2.3;
+    } else if grade == "C" {
+        points = 2.0;
+    } else if grade == "C-" {
+        points = 1.7;
+    } else if grade == "D+" {
+        points = 1.3;
+    } else if grade == "D" {
+        points = 1.0;
+    } else if grade == "F" {
+        points = 0.0;
+    } else {
+        panic!("Invalid grade entered.");
```

Comment:
Ooo `panic!` is pretty cool B)

Reply:
I reckoned it's a little more advanced than most of the features seen in beginner rust code on the web, but its use seems to at least encourage the use of idioms, at the cost of complexity. A worthy cause.